### PR TITLE
[FLINK-16590] flink-oss-fs-hadoop: remove erroneous license references

### PR DESCRIPTION
**Pull request by aljoscha**
I checked the output of the maven shade plugin on `flink-oss-fs-hadoop`, combined that with `flink-fs-hadoop-shaded`, sorted, cleaned it up. Then took this as the basis and cleaned up the `NOTICE` file and licenses.

@zentol Could you please review this?

cc @GJL

### CODE REVIEW

File: flink-filesystems/flink-oss-fs-hadoop/src/main/resources/META-INF/NOTICE [hunk 1]
```diff
@@ -6,104 +6,43 @@ The Apache Software Foundation (http://www.apache.org/).
...
-- org.eclipse.jetty:jetty-security:9.3.19.v20170502
-- org.eclipse.jetty:jetty-server:9.3.19.v20170502
-- org.eclipse.jetty:jetty-servlet:9.3.19.v20170502
-- org.eclipse.jetty:jetty-util:9.3.19.v20170502
-- org.eclipse.jetty:jetty-webapp:9.3.19.v20170502
-- org.eclipse.jetty:jetty-xml:9.3.19.v20170502
-- org.xerial.snappy:snappy-java:1.1.4
+ - com.aliyun.oss:aliyun-sdk-oss:jar:3.4.1
+ - com.aliyun:aliyun-java-sdk-core:jar:3.4.0
+ - com.aliyun:aliyun-java-sdk-ecs:jar:4.2.0
```
Comment (zentol) [inline flink-filesystems/flink-oss-fs-hadoop/src/main/resources/META-INF/NOTICE:11]: please remove the leading space

**Comment by aljoscha** [comment]
Thanks, I was unclear about this because the previous `NOTICE` file had all the dependencies from  `flink-fs-hadoop-shaded` also in there.

This `NOTICE` file was way to big... 😅

**Comment by zentol** [comment]
The `NOTICE` file was so big because previously this module was bundling `hadoop-common`. This was changed in FLINK-15835, but the licensing wasn't adjusted.

File: flink-filesystems/flink-oss-fs-hadoop/src/main/resources/META-INF/NOTICE [hunk 2]
```diff
@@ -6,117 +6,24 @@ The Apache Software Foundation (http://www.apache.org/).
...
+- com.aliyun:aliyun-java-sdk-ram:jar:3.0.0
+- com.aliyun:aliyun-java-sdk-sts:jar:3.0.0
+- com.aliyun:aliyun-java-sdk-ecs:jar:4.2.0
 
 This project bundles the following dependencies under the Common Development and Distribution License (CDDL) 1.0.
 See bundled license files for details.
 
-- javax.ws.rs:jsr311-api:1.1.1
-- javax.xml.stream:stax-api:1.0-2
 - stax:stax-api:1.0.1
```
Comment (zentol) [inline flink-filesystems/flink-oss-fs-hadoop/src/main/resources/META-INF/NOTICE:24]: According to https://mvnrepository.com/artifact/stax/stax-api/1.0.1 and the pom this is Apache licensed. Only `javax.xml.stream:stax-api` uses the CDDL license.

Comment (aljoscha) [reply flink-filesystems/flink-oss-fs-hadoop/src/main/resources/META-INF/NOTICE:24]: I see, then this was incorrect in the `NOTICE` file before. I'll update.

Do you know why it was in this section before?

Comment (zentol) [reply flink-filesystems/flink-oss-fs-hadoop/src/main/resources/META-INF/NOTICE]: My guess it that they assumed that all `stax-api` dependencies have the same license.
While I double-checked this I got confused as well as I found an entry on one [page](https://enterprise.dejacode.com/components/public/StAX+API/1.0-2/#essentials) listing it as CDDL, with only a tiny detail (the version `1.0-2` in the url) indicating that it is actually the `javax.xml.stream` one.

File: flink-filesystems/flink-oss-fs-hadoop/src/main/resources/META-INF/NOTICE [hunk 3]
```diff
@@ -6,117 +6,20 @@ The Apache Software Foundation (http://www.apache.org/).
...
-- javax.ws.rs:jsr311-api:1.1.1
-- javax.xml.stream:stax-api:1.0-2
+- org.apache.hadoop:hadoop-aliyun:jar:3.1.0
+- com.aliyun.oss:aliyun-sdk-oss:jar:3.4.1
+- org.apache.httpcomponents:httpclient:jar:4.5.3
+- org.apache.httpcomponents:httpcore:jar:4.4.6
+- commons-logging:commons-logging:jar:1.1.3
+- commons-codec:commons-codec:jar:1.10
+- org.codehaus.jettison:jettison:jar:1.1
 - stax:stax-api:1.0.1
```
Comment (zentol) [inline flink-filesystems/flink-oss-fs-hadoop/src/main/resources/META-INF/NOTICE:16]: sorting is broken

**Comment by aljoscha** [comment]
Thanks for the review!

**Pull request closed without merging**